actions/setup-crib-environment/action.yml

```diff
@@ -0,0 +1,169 @@
+name: "Setup CRIB Environment"
+description:
+  "Composite action for deploying a CRIB, setting up GAP, Nix, and deploying to
+  an ephemeral environment."
+
+inputs:
+  api-gateway-host:
+    description: API Gateway host for K8s
+    required: true
+  aws-region:
+    description: AWS region
+    required: true
+  aws-role-arn:
+    description: AWS Role ARN for CRIB
+    required: true
+  devspace-ingress-cidrs:
+    default: 0.0.0.0/0
+    description: DevSpace ingress CIDRs
+    required: false
+  disable-environment-teardown:
+    default: "false"
+    description: Disable namespace deletion for debugging purposes.
+    required: false
+  ecr-private-registry:
+    default: ""
+    description: ECR private registry account ID
+    required: false
+  ecr-private-registry-stage:
+    default: ""
+    description: ECR private registry account ID in Staging
+    required: false
+  image-tag:
+    default: latest
+    description: Docker image tag for the product
+    required: false
+  ingress-base-domain:
+    description: DevSpace ingress base domain
+    required: true
+  k8s-cluster-name:
+    description: K8s cluster name
+    required: true
+  kube-cache-dir:
+    default: /dev/null
+    description: Configure kube cache dir
+    required: false
+  ns-ttl:
+    default: 1h
+    description: Namespace TTL, used by Kyverno for cleanup
+    required: false
+  product:
+    default: core
+    description: Product name
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check Required Inputs and Validate Product
+      shell: bash
+      run: |
+        # Validate required inputs
+        [[ "${{ inputs.aws-role-arn }}" ]] || { echo "aws-role-arn input is empty" ; exit 1; }
+        [[ "${{ inputs.api-gateway-host }}" ]] || { echo "api-gateway-host input is empty" ; exit 1; }
+        [[ "${{ inputs.aws-region }}" ]] || { echo "aws-region input is empty" ; exit 1; }
+        [[ "${{ inputs.ingress-base-domain }}" ]] || { echo "ingress-base-domain input is empty" ; exit 1; }
+        [[ "${{ inputs.k8s-cluster-name }}" ]] || { echo "k8s-cluster-name input is empty" ; exit 1; }
+
+        # Validate product input
+        case "${{ inputs.product }}" in
+          "core"|"ccip")
+            echo "Product input is valid: ${{ inputs.product }}"
+            ;;
+          *)
+            echo "Invalid product input: ${{ inputs.product }}. Must be 'abc' or 'bcd'."
+            exit 1
+            ;;
+        esac
+
+        echo "All required inputs are provided and valid."
+
+    - name: Checkout the repository
+      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      with:
+        repository: "smartcontractkit/crib"
```

Comment:
we will need GATI for this no?

Reply:
I think you are right.

Reply:
Add an input for Github token as Chad suggested. Please see the comment below.